### PR TITLE
cilium: encryption, add debug statements to improve logs

### PR DESCRIPTION
**Pull request by jrfastab**
Adding logging for encryption/decryption control plane failures so we
can track these down.

Signed-off-by: John Fastabend <john.fastabend@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7299)

### CODE REVIEW

**Comment by jrfastab** [comment]
test-me-please

File: pkg/datapath/linux/node.go [hunk 1]
```diff
@@ -426,6 +426,14 @@ func (n *linuxNodeHandler) NodeUpdate(oldNode, newNode node.Node) error {
 }
 
 func (n *linuxNodeHandler) enableIPsec(newNode *node.Node) {
+	upsertIPsecLog := func(err error, spec string, loc, rem *net.IPNet) {
+		if err != nil {
+			log.WithError(err).Errorf("Upsert %s local %s remote %s", spec, loc, rem)
```
Comment (joestringer) [inline pkg/datapath/linux/node.go]: Usually we include additional values through `log.WithFields()`, which allows log scrapers to easily find the terms:

```
scopedLog := log.WithFields(logrus.fields{
    logfields.Reason: spec,
    "local-ip": loc,
    "remote-ip": rem,
});
if err != nil {
    scopedLog.WithError(err).Error("IPSec enable failed")
} else {
    scopedLog.Debug("Enabled IPsec")
}
```

File: pkg/datapath/linux/node.go [hunk 2]
```diff
@@ -438,19 +446,16 @@ func (n *linuxNodeHandler) enableIPsec(newNode *node.Node) {
...
 				ipsecLocal := &net.IPNet{IP: n.nodeAddressing.IPv4().Router(), Mask: n.nodeAddressing.IPv4().AllocationCIDR().Mask}
 				ipsecRemote := &net.IPNet{IP: ciliumInternalIPv4, Mask: newNode.IPv4AllocCIDR.Mask}
 				n.replaceNodeIPSecOutRoute(new4Net)
 				err := ipsec.UpsertIPSecEndpoint(ipsecLocal, ipsecRemote, linux_defaults.IPSecEndpointSPI, ipsec.IPSecDirOut)
-				if err != nil {
-					log.WithError(err).Errorf("Upsert local %s remote %s", ipsecLocal, ipsecRemote)
-				} else {
-					log.Debugf("Upsert (success) local %s remote %s", ipsecLocal, ipsecRemote)
-				}
+				upsertIPsecLog(err, "", ipsecLocal, ipsecRemote)
```
Comment (joestringer) [inline pkg/datapath/linux/node.go]: A couple of these pass in `""` as the spec, how would you tell them apart? Can we form better messages?

**Review by joestringer — CHANGES_REQUESTED** [review]
Minor request for better field handling, and it'd be good to get proper error messages so if a user ends up hitting the error, it provides more context about how they are affected.

Eg, rather than this:
```
log ERR | msg="Upsert ipsec"
```
Something more like:
```
log ERR | msg="Upsert failed, traffic may not be encrypted to this node"
```

File: pkg/datapath/linux/node.go (hunk 1 above)
Comment (jrfastab) [reply pkg/datapath/linux/node.go]: Yeah I sort of reinvented a poor version of WithFIelds will use standard approach.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
```
06:59:50  Timed out while waiting for the machine to boot. This means that
06:59:50  Vagrant was unable to communicate with the guest machine within
06:59:50  the configured ("config.vm.boot_timeout" value) time period.
06:59:50  
06:59:50  If you look above, you should be able to see the error(s) that
06:59:50  Vagrant had when attempting to connect to the machine. These errors
06:59:50  are usually good hints as to what may be wrong.
06:59:50  
06:59:50  If you're using a custom box, make sure that networking is properly
06:59:50  working and you're able to connect to the machine. It is a common
06:59:50  problem that networking isn't setup properly in these boxes.
06:59:50  Verify that authentication configurations are also setup properly,
06:59:50  as well.
06:59:50  
```

**Comment by jrfastab** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Pull request merged**